### PR TITLE
Strictly type db ids

### PR DESCRIPTION
**Pull request by dessalines**
To be merged after #1428

### CODE REVIEW

**Review by Nutomic — APPROVED** [review]
Very good! Is this also gonna complain if you pass in an i32 instead of UserID?

**Comment by dessalines** [comment]
Yep, that's the main thing. I didn't type *every* column obvi, just a few of the most popular ones with lots of foreign key constraints. But yep it does throw errors if you use the wrong type!

File: crates/db_schema/src/lib.rs [hunk 1]
```diff
@@ -18,6 +22,45 @@ use url::Url;
...
+pub struct PostId(pub i32);
+
+impl fmt::Display for PostId {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    write!(f, "{}", self.0)
+  }
+}
+
+#[derive(Debug, Copy, Clone, Hash, Eq, PartialEq, Serialize, Deserialize, DieselNewType)]
+pub struct PersonId(pub i32);
```
Comment (dessalines) [inline crates/db_schema/src/lib.rs:35]: This is the only really important change worth looking at here, cause everything else is boilerplate. These fields add `DieselNewType`, which implements a lot of the `FromSql` / `ToSql` things for you.

Comment (Nutomic) [reply crates/db_schema/src/lib.rs:35]: I hope this doesnt increase compilation time too much. Will benchmark it again once we are close to releasing.

**Pull request merged**